File: app/src/main/java/wjy/yo/ereader/entity/dict/WordRank.java
package wjy.yo.ereader.entity.dict;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;


@Entity(tableName = "dict_word_rank", indices = {@Index(value = {"word", "name"}, unique = true)})
public class WordRank {
    @PrimaryKey(autoGenerate = true)
    @NonNull
    private Integer id;

    @ForeignKey(entity = Dict.class, parentColumns = "word", childColumns = "word", onDelete = ForeignKey.CASCADE)
    private String word;

    private String name;

    private Integer rank;

    @NonNull
    public Integer getId() {
        return id;
    }

    public void setId(@NonNull Integer id) {
        this.id = id;
    }

    public String getWord() {
        return word;
    }

    public void setWord(String word) {
        this.word = word;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getRank() {
        return rank;
    }

    public void setRank(Integer rank) {
        this.rank = rank;
    }
}

File: app/src/main/java/wjy/yo/ereader/entity/dict/WordRank.kt
package wjy.yo.ereader.entity.dict

import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey
import android.arch.persistence.room.Index
import android.arch.persistence.room.PrimaryKey


@Entity(tableName = "dict_word_rank", indices = arrayOf(Index(value = arrayOf("word", "name"), unique = true)))
class WordRank {
    @PrimaryKey(autoGenerate = true)
    var id: Int? = 0

    @ForeignKey(entity = Dict::class, parentColumns = arrayOf("word"), childColumns = arrayOf("word"), onDelete = ForeignKey.CASCADE)
    var word: String? = null

    var name: String? = null

    var rank: Int? = null
}
